### PR TITLE
Add JavaScript for Google Tag Manager

### PR DESCRIPTION
**Pull request by settermjd**
The intent is to be able to use the Google Tag Manager in tandem with the Google Search Console to help manage site search results. This addition should enable Google Tag Manager support in the production docs.

### CODE REVIEW

File: src/partials/footer-scripts.hbs [hunk 1]
```diff
@@ -5,3 +5,8 @@
 <script src="{{uiRootPath}}/js/vendor/highlight.js"></script>
 <script>hljs.initHighlighting()</script>
 <link href="https://fonts.googleapis.com/css?family=Raleway|Roboto" rel="stylesheet">
+
+<!-- Google Tag Manager (noscript) -->
+<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-NXSDZWN"
+height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+<!-- End Google Tag Manager (noscript) -->
```
Comment (hurradieweltgehtunter) [inline src/partials/footer-scripts.hbs]: Script should be included right after <body> tag

**Comment by tboerger** [comment]
This snippet should ONLY be enabled for the documentation. As this theme is used on multiple places it should be placed in a partial which only gets content within the docs repo or it should only be enabled if some variable had been defined. But the current PR is a blocker :-1:.

**Comment by settermjd** [comment]
@hurradieweltgehtunter / @tboerger, feedback has been incorporated.

**Comment by settermjd** [comment]
@tboerger, what's required to set the `IS_DOCS` environment variable when the docs are deployed?

File: src/layouts/default.hbs [hunk 2]
```diff
@@ -24,4 +24,10 @@
 {{> main}}
 {{> footer}}
   </body>
+{{#if env.IS_DOCS}}
+  <!-- Google Tag Manager (noscript) -->
+  <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-NXSDZWN"
+  height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+  <!-- End Google Tag Manager (noscript) -->
+{{/if}}
```
Comment (hurradieweltgehtunter) [inline src/layouts/default.hbs]: Sorry if I was unclear. This part needs to be right after the **opening** body tag, not the closing one.

**Comment by tboerger** [comment]
I think best would be really prepare an empty partial here and fill it within the docs repo.

File: src/layouts/default.hbs (hunk 2 above)
Comment (settermjd) [reply src/layouts/default.hbs]: Np.

Comment (settermjd) [reply src/layouts/default.hbs]: Done.

**Comment by settermjd** [comment]
> I think best would be really prepare an empty partial here and fill it within the docs repo.

The code's been refactored to a partial and renders a configurable ID, if set in the environment. That what you're after?

**Comment by tboerger** [comment]
IMHO it's better to provide empty partials which get content within the docs repo as part of the overlay folder.

**Comment by settermjd** [comment]
All this one needs is an id, which can be provided by an environment variable, and is set to only show if said variable is set. More work seems like overkill.

**Comment by tboerger** [comment]
But this variable always got to be defined when building the page which lead to issues in the past, that's why I asked to put it tinto the repo that really needs that and not into a repo that is shared accross multiple Antora pages.

**Pull request closed without merging**